sklearn/decomposition/_dict_learning.py

```diff
@@ -1630,6 +1631,26 @@ class DictionaryLearning(_BaseSparseCoding, BaseEstimator):
     0.07...
     """
 
+    _parameter_constraints = {
+        'n_components': [Interval(Integral, 0, None, closed='neither'), None],
+        'alpha': [Interval(Real, 0, 1, closed='left')],
+        'max_iter': [Interval(Integral, 1, None, closed='neither')],
+        'tol': [Interval(Real, 0, None, closed='left')],
+        'fit_algorithm': [StrOptions({'lars', 'cd'})],
+        'transform_algorithm': [StrOptions({'lasso_lars', 'lasso_cd', 'lars', 'omp', 'threshold'})],
+        'transform_n_nonzero_coefs': [Interval(Integral, 0, None, closed='neither'), None],
+        'transform_alpha': [Interval(Real, 0, 1, closed='left'), None],
+        'n_jobs': [Interval(Integral, -1, None, closed='neither'), None],
+        'code_init': ['array-like', None],
+        'dict_init': ['array-like', None],
+        'verbose': ['verbose'],
+        'split_sign': [Interval(Integral, 0, 1, closed='neither'), 'boolean'],
+        'random_state': ['random_state'],
+        'positive_code': [Interval(Integral, 0, 1, closed='neither'), 'boolean'],
+        'positive_dict': [Interval(Integral, 0, 1, closed='neither'), 'boolean'],
+        'transform_max_iter': [Interval(Integral, 1, None, closed='neither')],
```

Comment:
@ShisuiUzumaki - A tip from a fellow contributor on this meta-issue: this code is currently failing the linting check. Scikit-learn uses black formatting, so you will have to change the single quotes to double quotes here and address any long lines to pass this test. You can see exactly what needs to be changed by looking at the output of the job [here](https://dev.azure.com/scikit-learn/scikit-learn/_build/results?buildId=45923&view=logs&jobId=32e2e1bb-a28f-5b18-6cfc-3f01273f5609&j=32e2e1bb-a28f-5b18-6cfc-3f01273f5609&t=fc67071d-c3d4-58b8-d38e-cafc0d3c731a).

I would also suggest that you set up pre-commit locally, as that will save you time on these things. Just run the following from the top-level of the repo and the formatting checks will run locally on your next commit:
```shell
pip install pre-commit
pre-commit install
```

Reply:
Thanks for advice I will do that

Reply:
Regarding this, you can also install `black` and run it manually. I tend to prefer that, but I also use `pre-commit` since it is a life-saver! If you do go for manually using `black`, ensure that it is version `22.3.0`

Reply:
Okay